libcxx/test/std/utilities/smartptr/adapt/inout_ptr/inout_ptr.general.pass.cpp

```diff
@@ -0,0 +1,196 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// <memory>
+
+// [inout.ptr], function template inout_ptr
+// template<class Pointer = void, class Smart, class... Args>
+//   auto inout_ptr(Smart& s, Args&&... args);                 // since c++23
+
+#include <cassert>
+#include <memory>
+#include <utility>
+
+#include "../types.h"
+
+void test_replace_int_p() {
```

Comment:
Please include a plain-english comment explaining these test cases. It's OK to put them in separate functions, but the function name isn't really enough to convey what's being tested here, a plain english comment is helpful.

Reply:
Done! Thanks that was a hard one. Please check my "plain English comments" if they make any sense.